### DISCUSSION TITLE
Discussion about the XC browser feature and Debian maintainer

### DISCUSSION THREAD

**Discussion by ksaadDE** [keepassxreboot/keepassxc#10742] [Ideas]
Sadly, I just got informed about the issue 10725,the message by the German Debian KC package maintainer working for the company behind Ubuntu (Canonical Ltd - conflict of interest?).   When I wanted to comment I saw it's locked.

So we have that:
- https://github.com/keepassxreboot/keepassxc/issues/10725
- https://news.ycombinator.com/item?id=40320166
- https://bugs.debian.org/cgi-bin/bugreport.cgi?bug=953529

As user of all distros, especially Debian and Arch I am absolutely confused about that behave. The statement given by him is:
> I'm afraid that's not going to happen. It was a **mistake to ship with all plugins built by default**. This will be painful for a year as **users annoyingly do not read the NEWS** files they should be reading but there's little that can be done about that.
 It is **our responsibility to our users to provide them the most secure option** possible as the default. **All of these features** are superfluous and **do not** really **belong** in a **local password database manager**, these developments are all utterly misguided.
 **Users who need this crap can install the crappy version** but obviously this increases the **risk of drive-by contributor attacks**
 
So the maintainer not only insults the users who are using appropriate opt-in features without any valid ground. He is inable to prove that there are any actual security risks or security issues involved. He absolutely bases all on a privacy NOT SECURITY concern on the Debian Bug Tracker. It is about a few icons (or anything like that) being loaded. Despite the issue talks about the broken browser integration. It is also unclear what the maintainer is exactly talking about when he says (by-default disabled) "plugins". 

There is **no obvious security risk**. There might be a privacy risk, of course, when a password manager connects to somewhere. Every other application does that too. Especially those on Debian. So why does the maintainer speaks that hypocrisy out? Because he wants to by pass the need to justify his stance.

Another **point highlighted by the community** that people need those features. Features are industry standards. The XC project seemingly disables them by default and goes for opt-in, which is already the most secure version of it. If Debian does not supply them, the users will neither consider Debian in a good way nor the XC project, causing distrust and damage for us all.

To be honest: From a security and user pov this is inacceptable. 

Maintainers randomly deciding to disable/enable flags, lying about the reason, making up terms,  breaking important core features used by all, causing bug reports to a open project without proper communication. Finally, after being invited in the discussion, he's not providing reasons, only excuses and even really bad ones. On top insulting the users is more than disappointing, disgusting, disrespecting and gives the Debian project a really weird shade!

This is exactly why only a few trust the Linux and (Fully) Open Source Community and people like this maintainer destroy the trust that has been shaped over the many years. Just for his toxic, individual opinion - sorry for the rant! 

/E: And yes, nobody reads your damn news, if they are written on a board from the 2000s, wake up - you're living in 2024. 
/E1: As a German I am expecting more of my people, not this crap. So I am pissed off three times at once.  

Regards

**Comment by ksaadDE** [comment]
"drive-by contributor attacks" is a undefined term! I tried to look it up, but there is no literature about this, nor any wiki or forum posts mentioning it. So I am assuming it is made up. So what does that even mean?  And it should be mean something, if it is used in a justification for actions.

![grafik](https://github.com/keepassxreboot/keepassxc/assets/37943746/a7b06450-f2d6-407c-bb9c-44bb3369f501)
![grafik](https://github.com/keepassxreboot/keepassxc/assets/37943746/d18b341f-adb1-4147-81ff-11f843e5c8d2)
![grafik](https://github.com/keepassxreboot/keepassxc/assets/37943746/5d72eb2f-e8c6-4554-815a-e6dba2fad4d4)

**Comment by TukanDev** [comment]
He distrusts internet capability so much he straight up forced pushed an updated to people just wanting their password manager to work not spend time realizing "oh shit maintainer of that package is insane and stripped most basic features like autofill which DONT EVEN USE INTERNET CONECTIVITY".
As many people disscused about this issue there should have been keypassxc-minimal instead of breaking thing many people use... 
You are free to package the software, you are also free to modify it and build it with build flags that exist however you see fit BUT BY NO MEANS you should try to act smarter and try to package it with disabled defaults as an updated for existing users, or try to be smart and think you know how to package a software. If maintainer tells you not to package it that way as you dont know how to do it properly just dont do it. This is why bottles developers went apeshit since people came opening issues about broken packaging instead of using official distributed package in flatpak so developers got fed up with it. You are just a package maintainer you have knowledge but you are not smarter to know how to package every software properly. Same how steam on snap store incident they tried to be smart instead made valve go insane with issue reports of snap steam...
If you do not need network features build local copy for yourself dont push it in a package you maintain to enforce your opinion and agenda, you have no power of that as you do not know better then author of the software you are packaging.

**Comment by phoerious** [comment]
As I understood Julian, he will now provide two versions and a transitional package instead of straight up downgrading everybody.

**Reply by ksaadDE** [reply]
Interesting! I have not read that. Still the thought and move in the first place is damaging enough, yet alone the insulting comment by him to the user base.

**Comment by ksaadDE** [comment]
fyi @julian-klode

**Comment by EmperorArthur** [comment]
As a heads up, [this video](https://www.youtube.com/watch?v=9AV1Q-JfHpI) about the issue went out 18 hours ago, and is currently sitting at 20k views and ~600 comments.

It's what brought it to my attention, and the attention of many others.

**Reply by ksaadDE** [reply]
Thank you very much. Great video.
